src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs

```diff
@@ -430,12 +430,16 @@ private static void TestTypeInfoImmutability<T>(JsonTypeInfo<T> typeInfo)
                 Assert.Null(property.ShouldSerialize);
                 Assert.Null(typeInfo.NumberHandling);
 
+                Assert.Throws<InvalidOperationException>(() => property.AttributeProvider = property.AttributeProvider);
                 Assert.Throws<InvalidOperationException>(() => property.CustomConverter = property.CustomConverter);
                 Assert.Throws<InvalidOperationException>(() => property.Name = property.Name);
                 Assert.Throws<InvalidOperationException>(() => property.Get = property.Get);
                 Assert.Throws<InvalidOperationException>(() => property.Set = property.Set);
                 Assert.Throws<InvalidOperationException>(() => property.ShouldSerialize = property.ShouldSerialize);
                 Assert.Throws<InvalidOperationException>(() => property.NumberHandling = property.NumberHandling);
+                Assert.Throws<InvalidOperationException>(() => property.Order = property.Order);
+                Assert.Throws<InvalidOperationException>(() => property.IsExtensionData = property.IsExtensionData);
+                Assert.Throws<InvalidOperationException>(() => property.IsRequired = property.IsRequired);
```

Comment:
Can these tests be re-written with reflection so we avoid missing some props as shown here?